### PR TITLE
miniupnpc: update to @2.2.3

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Update `miniupnpc` to @2.2.3

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.7
Xcode 12.4

macOS 10.6.8 Server
Xcode 3.2.6

macOS 10.6 PPC (10A190)
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by barracuda156** [comment]
@mascguy @kencu These is no maintainer. If no one objects, could someone push it?

**Comment by barracuda156** [comment]
> I'm sorry I didn't see this before.
> 
> When you are building things in MacPorts using cmake, a very large amount of MacPorts configuration is set up using the cmake PortGroup.
> 
> So instead of adding the cmake build dependency, and probably instead of manually changing the install path of the include files, what you do is use the PortGroup
> 
> ```
> PortGroup cmake 1.1
> ```
> 
> give that a try instead of your depends_build and patch, and let's see how it works out!

@kencu Thank you for responding. I tried, but it failed without a patch:
```
Install the project...
/opt/local/bin/cmake -P cmake_install.cmake
-- Install configuration: "MacPorts"
-- Installing: /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_net_miniupnpc/miniupnpc/work/destroot/opt/local/lib/cmake/miniupnpc/miniupnpc-private.cmake
-- Installing: /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_net_miniupnpc/miniupnpc/work/destroot/opt/local/lib/libminiupnpc.a
-- Installing: /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_net_miniupnpc/miniupnpc/work/destroot/opt/local/lib/cmake/miniupnpc/libminiupnpc-static.cmake
-- Installing: /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_net_miniupnpc/miniupnpc/work/destroot/opt/local/lib/cmake/miniupnpc/libminiupnpc-static-macports.cmake
-- Installing: /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_net_miniupnpc/miniupnpc/work/destroot/opt/local/lib/libminiupnpc.2.2.3.dylib
-- Installing: /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_net_miniupnpc/miniupnpc/work/destroot/opt/local/lib/libminiupnpc.17.dylib
-- Installing: /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_net_miniupnpc/miniupnpc/work/destroot/opt/local/lib/libminiupnpc.dylib
-- Installing: /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_net_miniupnpc/miniupnpc/work/destroot/opt/local/lib/cmake/miniupnpc/libminiupnpc-shared.cmake
-- Installing: /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_net_miniupnpc/miniupnpc/work/destroot/opt/local/lib/cmake/miniupnpc/libminiupnpc-shared-macports.cmake
CMake Error at cmake_install.cmake:131 (file):
  file INSTALL cannot find
  "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_net_miniupnpc/miniupnpc/work/miniupnpc-2.2.3/miniupnpc.h":
  No such file or directory.


make: *** [install/fast] Error 1
make: Leaving directory `/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_net_miniupnpc/miniupnpc/work/build'
Command failed:  cd "/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_net_miniupnpc/miniupnpc/work/build" && /usr/bin/make -w install/fast INSTALLPREFIX=/opt/local CC="/usr/bin/gcc-4.2 -Os -DNDEBUG -I/opt/local/include -arch ppc" DESTDIR=/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_net_miniupnpc/miniupnpc/work/destroot 
Exit code: 2
```

**Comment by reneeotten** [comment]
@barracuda156 please do not remove the PR checklist when you submit a PR. Instead do check off the thing you actually did (and ideally you should do of course all of them) so that it is clear how well tested things are.

**Comment by barracuda156** [comment]
> @barracuda156 please do not remove the PR checklist (here and in others) when you submit a PR. Instead do check off the thing you actually did (and ideally you should do of course all of them) so that it is clear how well tested things are.

@reneeotten Thank you, I won’t be removing it. (Here it is a mere update of the version with no tweaks aside of fixing includes linking, therefore I thought it is redundant. The single patch is trivial.)

**Comment by barracuda156** [comment]
@reneeotten @mascguy @kencu If everyone happy, maybe we could push it?

File: net/miniupnpc/Portfile [hunk 1]
```diff
@@ -1,11 +1,12 @@
 # -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:filetype=tcl:et:sw=4:ts=4:sts=4
 
 PortSystem          1.0
+PortGroup			cmake 1.1
```
Comment (reneeotten) [inline net/miniupnpc/Portfile:4]: please align this properly

File: net/miniupnpc/Portfile [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
 
 name                miniupnpc
 epoch               2
-version             2.2.2
-revision            0
+version             2.2.3
+revision            1
```
Comment (reneeotten) [inline net/miniupnpc/Portfile:9]: leave the revison at zero since you increased the version.

**Review by reneeotten — CHANGES_REQUESTED** [review]
Not sure about the patch - seems unlikely that upstream messed this up...

**Comment by barracuda156** [comment]
> Not sure about the patch - seems unlikely that upstream messed this up...

@reneeotten The patch is actually from upstream: https://github.com/miniupnp/miniupnp/commit/f82b0563a72e33800f5918ead68c41bcda1fd4c1
Referred by: https://github.com/miniupnp/miniupnp/issues/574#issuecomment-1086778404

**Comment by reneeotten** [comment]
> > Not sure about the patch - seems unlikely that upstream messed this up...
> 
> 
> 
> @reneeotten The patch is actually from upstream: https://github.com/miniupnp/miniupnp/commit/f82b0563a72e33800f5918ead68c41bcda1fd4c1
> 
> Referred by: https://github.com/miniupnp/miniupnp/issues/574#issuecomment-1086778404


then please add that link at the top of the patchfile to reflect that fact.

**Comment by barracuda156** [comment]
> then please add that link at the top of the patchfile to reflect that fact.

@reneeotten Three requested changes made now.

**Comment by barracuda156** [comment]
@reneeotten @mascguy @kencu I think this should be ready to be pushed into the master.

**Review by mascguy — APPROVED** [review]

**Comment by jmroot** [comment]
> [miniupnp/miniupnp#574 (comment)](https://github.com/miniupnp/miniupnp/issues/574#issuecomment-1086778404)

Lack of a .pc file as mentioned in that issue is a serious problem. Some dependents are now unable to find the library.

**Pull request merged**